pkg/gather/clusterconfig/clusterconfig.go

```diff
@@ -202,19 +205,48 @@ func GatherClusterOperators(i *Gatherer) func() ([]record.Record, []error) {
 				continue
 			}
 			for _, namespace := range namespacesForOperator(&item) {
+				if namespaceEventsCollected.Has(namespace) {
+					continue
+				}
 				pods, err := i.coreClient.Pods(namespace).List(metav1.ListOptions{})
 				if err != nil {
 					klog.V(2).Infof("Unable to find pods in namespace %s for failing operator %s", namespace, item.Name)
 					continue
 				}
-				for i := range pods.Items {
-					if isHealthyPod(&pods.Items[i], now) {
+				for j := range pods.Items {
+					pod := &pods.Items[j]
+					if isHealthyPod(pod, now) {
 						continue
 					}
-					records = append(records, record.Record{Name: fmt.Sprintf("config/pod/%s/%s", pods.Items[i].Namespace, pods.Items[i].Name), Item: PodAnonymizer{&pods.Items[i]}})
-				}
-				if namespaceEventsCollected.Has(namespace) {
-					continue
+					records = append(records, record.Record{Name: fmt.Sprintf("config/pod/%s/%s", pod.Namespace, pod.Name), Item: PodAnonymizer{pod}})
+
+					// Fetch previous and current pod logs
+					for _, previous := range []bool{true, false} {
+						for _, container := range pod.Spec.Containers {
+							currentOrPrevious := fmt.Sprintf("%s_current.log", container.Name)
+							if previous {
+								currentOrPrevious = fmt.Sprintf("%s_previous.log", container.Name)
+							}
+
+							req := i.coreClient.Pods(namespace).GetLogs(pod.Name, &corev1.PodLogOptions{Previous: previous, TailLines: &logTailLines, Container: container.Name})
+							readCloser, err := req.Stream()
+							if err != nil {
+								klog.V(2).Infof("Failed to fetch %s for %s pod in namespace %s for failing operator %s: %q", currentOrPrevious, pod.Name, namespace, item.Name, err)
+								continue
+							}
+
+							defer readCloser.Close()
+
+							buf := bytes.NewBufferString("")
+							_, err = io.Copy(buf, readCloser)
+							if err != nil {
+								klog.V(2).Infof("Failed to write %s for %s pod in namespace %s for failing operator %s: %q", currentOrPrevious, pod.Name, namespace, item.Name, err)
+								continue
+							}
```

Comment:
Would it be possible to move this to a new function pls? Same `collectContainerLogs` as we have in 4.5 and master. There's also some extra logic calculating the buffer size in 4.5 and master.

Just for the sake of smoother backports.

Reply:
Backported `collectContainerLogs` w/ size limitation